x-pack/auditbeat/module/system/socket/_meta/data.json

```diff
@@ -0,0 +1,46 @@
+{
+    "@timestamp": "2017-10-12T08:05:34.853Z",
+    "agent": {
+        "hostname": "host.example.com",
+        "name": "host.example.com"
+    },
+    "event": {
+        "action": "socket_opened",
+        "dataset": "socket",
+        "module": "system",
+        "type": "event"
+    },
+    "process": {
+        "name": "lynx",
+        "pid": 12016
+    },
+    "system": {
+        "socket": {
+            "network": {
+                "type": "ipv4",
+                "direction": "outgoing"
+            },
+            "state": "CLOSE-WAIT",
+            "user": {
+                "uid": 1000,
+                "name": "vagrant"
+            },
+            "source": {
+                "ip": "10.0.2.15",
+                "port": 56148
+            },
+            "destination": {
+                "ip": "216.58.213.68",
+                "port": 80
+            },
+            "process": {
+                "pid": 12016,
+                "name": "lynx"
+            }
+        }
+    },
+    "user": {
+        "id": 1000,
+        "name": "vagrant"
+    }
+}
```

Comment:
To fully utilize the ECS fields and minimize duplication I think the event should be structured as

```
{
  "@timestamp": "2017-10-12T08:05:34.853Z",
  "agent": {
    "hostname": "host.example.com",
    "name": "host.example.com"
  },
  "event": {
    "action": "socket_opened",
    "dataset": "socket",
    "module": "system",
    "type": "event"
  },
  "process": {
    "name": "lynx",
    "pid": 12016
  },
  "system": {
    "socket": {
      "state": "CLOSE-WAIT"
    }
  },
  "network": {
    "type": "ipv4",
    "direction": "outgoing"
  },
  "source": {
    "ip": "10.0.2.15",
    "port": 56148
  },
  "destination": {
    "ip": "216.58.213.68",
    "port": 80
  },
  "user": {
    "id": 1000,
    "name": "vagrant"
  }
}
```

Reply:
I like that. If we remove `state` (thread below) we could do away with the whole `system/socket` object - unusual (even unique?) for a metricset to be using only ECS fields, but I'm thinking why not.

Reply:
I don't see any problem with it not reporting any system.socket namespaced fields.